File: RealPropertyTaxProjection/frmLogin.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using Intangic.RPTP.Business.Exceptions;
using Intangic.RPTP.Business.Impl;
using Intangic.RPTP.Business.Interfaces;
using Intangic.RPTP.Domain;
using Intangic.RPTP.Factory;
using Intangic.Utils;

namespace RealPropertyTaxProjection
{
    public partial class frmLogin : Form
    {
        public IRealPropertyTaxProjectionService realPropertyTaxProjectionManager;

        public frmLogin()
        {
            InitializeComponent();
        }

        private void btnLogin_Click(object sender, EventArgs e)
        {
            LoginRequest request = null;
            LoginResponse response = null;

            request = new LoginRequest() {
                UserName = this.userName.Text,
                Password = this.password.Text };

            try {
                realPropertyTaxProjectionManager = BusinessDelegateFactory.GetInstance().GetRealPropertyTaxProjectionService();
                
                response = new LoginResponse();
                response = realPropertyTaxProjectionManager.Login(request); 
                if (response.IsLoginValid) {
                    this.DialogResult = DialogResult.OK;
                }
            }
            catch (RealPropertyTaxProjectionException ex) {
                FormHelper.MessageShow(ex.Message);
                this.userName.Focus();
            }
            catch (Exception ex) {
                FormHelper.MessageShow(ex.Message);
                this.userName.Focus();
            }
            finally {
                realPropertyTaxProjectionManager = null;
                response = null;
                request = null;
            }
        }
    }
}

File: RealPropertyTaxProjectionVB/frmLogin.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Imports Intangic.RPTP.Business.Exceptions
Imports Intangic.RPTP.Business.Impl
Imports Intangic.RPTP.Business.Interfaces
Imports Intangic.RPTP.Domain
Imports Intangic.RPTP.Factory
'Imports Intangic.Utils

Imports FormUtils = RealPropertyTaxProjectionVB.Intangic.Utils


Public Class frmLogin

    Public realPropertyTaxProjectionManager As IRealPropertyTaxProjectionService

    Private Sub login_Click(sender As Object, e As EventArgs) Handles login.Click
        Dim request As LoginRequest = Nothing
        Dim response As LoginResponse = Nothing


        request = New LoginRequest()
        With request
            .UserName = Me.userName.Text
            .Password = Me.password.Text
        End With


        Try
            realPropertyTaxProjectionManager = BusinessDelegateFactory.GetInstance().GetRealPropertyTaxProjectionService()

            response = New LoginResponse()
            response = realPropertyTaxProjectionManager.Login(request)
            If response.IsLoginValid Then
                Me.Hide()
                frmShowBuildingData.Show()
            End If

        Catch ex As RealPropertyTaxProjectionException
            FormUtils.FormHelper.MessageShow(ex.Message)
            Me.userName.Focus()

        Catch ex As Exception
            FormUtils.FormHelper.MessageShow(ex.Message)
            Me.userName.Focus()

        Finally
            realPropertyTaxProjectionManager = Nothing
            response = Nothing
            request = Nothing
        End Try
    End Sub

End Class

